File: app/src/main/java/de/sysitprep/app/data/model/Stats.kt
package de.sysitprep.app.data.model


data class UserStats(
    val totalAnswered: Int = 0,
    val correctAnswers: Int = 0,
    val completedSessions: Int = 0,
    val totalStudySeconds: Int = 0,
    val streakDays: Int = 0,
    val lastStudiedAt: Long = 0L
) {
    val accuracy: Float get() = if (totalAnswered == 0) 0f else correctAnswers.toFloat() / totalAnswered.toFloat()
    val accuracyPercent: Int get() = (accuracy * 100).toInt()
}


data class LernfeldProgress(
    val lernfeld: Int,
    val title: String = "Lernfeld $lernfeld",
    val totalQuestions: Int,
    val correctAnswers: Int
) {
    val progressPercent: Int get() = if (totalQuestions == 0) 0 else (correctAnswers.toFloat() / totalQuestions * 100).toInt()
}

data class QuizResult(
    val sessionId: Long,
    val totalQuestions: Int,
    val correctAnswers: Int,
    val wrongAnswers: Int,
    val skippedAnswers: Int,
    val scorePercent: Int,
    val passed: Boolean,
    val timeSeconds: Int,
    val questionResults: List<QuestionResult>
)

data class QuestionResult(
    val questionId: Long,
    val questionText: String,
    val options: List<String>,
    val correctIndices: List<Int>,
    val selectedIndices: List<Int>,
    val isCorrect: Boolean,
    val explanation: String
)

File: ExamForge-iOS/ExamForge/Models/Stats.swift
import Foundation

// MARK: - UserStats

struct UserStats {
    var totalAnswered: Int = 0
    var correctAnswers: Int = 0
    var completedSessions: Int = 0
    var totalStudySeconds: Int = 0
    var streakDays: Int = 0
    var lastStudiedAt: Date = Date(timeIntervalSince1970: 0)

    var accuracy: Float {
        totalAnswered == 0 ? 0.0 : Float(correctAnswers) / Float(totalAnswered)
    }
    var accuracyPercent: Int { Int(accuracy * 100) }
}

// MARK: - LernfeldProgress

struct LernfeldProgress: Identifiable {
    let id = UUID()
    let lernfeld: Int
    var title: String
    let totalQuestions: Int
    let correctAnswers: Int

    init(lernfeld: Int, title: String = "", totalQuestions: Int, correctAnswers: Int) {
        self.lernfeld = lernfeld
        self.title = title.isEmpty ? "Lernfeld \(lernfeld)" : title
        self.totalQuestions = totalQuestions
        self.correctAnswers = correctAnswers
    }

    var progressPercent: Int {
        totalQuestions == 0 ? 0 : Int(Float(correctAnswers) / Float(totalQuestions) * 100)
    }
}

// MARK: - QuizResult

struct QuizResult {
    let sessionId: Int64
    let totalQuestions: Int
    let correctAnswers: Int
    let wrongAnswers: Int
    let skippedAnswers: Int
    let scorePercent: Int
    let passed: Bool
    let timeSeconds: Int
    let questionResults: [QuestionResult]
}

struct QuestionResult: Identifiable {
    let id = UUID()
    let questionId: Int64
    let questionText: String
    let options: [String]
    let correctIndices: [Int]
    let selectedIndices: [Int]
    let isCorrect: Bool
    let explanation: String
}

// MARK: - Navigation Params

struct QuizParams: Identifiable {
    let id = UUID()
    let sessionType: SessionType
    let lernfeld: Int          // -1 means not applicable
    let fachrichtung: Fachrichtung
    let examType: ExamType
}

struct ResultParams: Identifiable {
    let id = UUID()
    let sessionId: Int64
    let originalParams: QuizParams
}

// MARK: - Lernfeld UI Item

struct LernfeldUiItem: Identifiable {
    let id = UUID()
    let lernfeld: Lernfeld
    let questionCount: Int
    let progress: Int
    let fachrichtung: Fachrichtung
}
